src/twisted/python/_url.py

```diff
@@ -7,787 +7,7 @@
 URL parsing, construction and rendering.
 """
 
-try:
-    from urlparse import urlsplit, urlunsplit
-    from urllib import quote as urlquote, unquote as urlunquote
-except ImportError:
-    from urllib.parse import (urlsplit, urlunsplit,
-                              quote as urlquote,
-                              unquote_to_bytes as urlunquote)
+from hyperlink._url import URL
```

Comment:
Hyperlink already has, this should be hyperlink.URL as documented on http://hyperlink.readthedocs.io/en/latest/api.html